src/coreclr/jit/liveness.cpp

```diff
@@ -2198,7 +2198,11 @@ bool Compiler::fgTryRemoveNonLocal(GenTree* node, LIR::Range* blockRange)
                 return GenTree::VisitResult::Continue;
             });
 
+#ifdef TARGET_ARM64
+            if (node->OperIs(GT_SELECTCC, GT_SETCC, GT_CINC, GT_CINCCC))
```

Comment:
`GT_CINC` should not be added here.

Reply:
I think you should be able to switch this to `node->OperConsumesFlags()`.

Reply:
Apparently not, looks like this needs to be reverted.

Reply:
Did you mean [this](https://github.com/dotnet/runtime/pull/82031/files#diff-12752e8259d2fec5afbfb574f8cadc381f7ed2c438d7d718e9deb742189d0514R2201) 🤔 ? It's changed to `node->OperConsumesFlags()` as suggested.

Reply:
Yes -- it is hitting asserts in the build:
```scala
Assertion failed '(node->gtPrev->gtFlags & GTF_SET_FLAGS) != 0' in 'System.Globalization.UmAlQuraCalendar:ConvertGregorianToHijri(System.DateTime,byref,byref,byref)' during 'Per block local var liveness' (IL size 277; hash 0xdced8152; FullOpts)
```

The problem is that certain opers that consume flags actually do not have the flags def right before. You can remove the assert and add it to the check, e.g. like this:
```scala
if (node->OperConsumesFlags() && node->gtPrev->gtSetFlags())
```

Reply:
Ooh, sorry. Didn't notice the assertion failures during builds in the CI. It's been fixed as you recommended and CI seems be happy too. I think, it didn't show up on our end as I built libraries in release mode.